pkg/framework/backoff.go

```diff
@@ -0,0 +1,32 @@
+package framework
+
+import (
+	"time"
+
+	"github.com/cenkalti/backoff"
+)
+
+const (
+	LongMaxWait  = 30 * time.Minute
+	ShortMaxWait = 2 * time.Minute
+)
+
+const (
+	LongMaxInterval  = 60 * time.Second
+	ShortMaxInterval = 5 * time.Second
+)
+
+func newExponentialBackoff(maxWait, maxInterval time.Duration) *backoff.ExponentialBackOff {
```

Comment:
We start using a backoff here for the retry logic. Right now retrying and timing out is kind of home grown and a bit mixed up with the usage of backoffs. Here we try to get this a bit more straight to make the code easier and prevent nasty side effects.